### ISSUE TITLE
Set envvar on device, so other tools can react to scrcpy

### ISSUE THREAD

**Issue by Dracon23** [Genymobile/scrcpy#119]
I use Tasker for device automation and other neat little gimmicks.

I wonder if it's possible to have scrcpy set an envvar when connecting to the device so I can have tasker do stuff like disabling display timeout and such.

**Comment by rom1v** [comment]
Hi,

How could an envvar help? (I don't know Tasker).

What about calling a wrapper script from the computer to do whatever you want?
Also see discussion: https://github.com/Genymobile/scrcpy/issues/96#issuecomment-375969218

**Comment by Dracon23** [comment]
I can (ab)use the anvvar as a trigger for tasker to run it's actions.

Right now the only way I have is determining if scrcpy is running if I intercept the USB-Debugging notification, but that's not... failureproof for reasons...
________________________________
Von: Romain Vimont <notifications@github.com>
Gesendet: Mittwoch, 11. April 2018 02:46
An: Genymobile/scrcpy
Cc: Dracon23; Author
Betreff: Re: [Genymobile/scrcpy] Set envvar on device, so other tools can react to scrcpy (#119)


Hi,

How could an envvar help? (I don't know Tasker).

What about calling a wrapper script from the computer to do whatever you want?
Also see discussion: #96 (comment)<https://github.com/Genymobile/scrcpy/issues/96#issuecomment-375969218>

**Comment by leaskovski** [comment]
@rom1v , the problem with using a wrapper script is that if the device is pulled without closing the scrcpy application first, and thus the wrapper script finishes without the device being connected, the code post run of scrcpy won't actually be run on the device,

**Comment by rom1v** [comment]
@leaskovski you're right, but in that case, the server application is killed too, so it may not directly send a message to "tasker". To detect termination, it would require something that is closed by the kernel on process termination, like a pipe or socket.
